app/models/concerns/event_concern.rb

```diff
@@ -2,13 +2,36 @@ module EventConcern
   extend ActiveSupport::Concern
 
   included do
-    after_destroy :raise_event, :prepend => true
+    after_destroy :raise_event_for_destroy, :prepend => true
   end
 
-  def raise_event
-    headers = Insights::API::Common::Request.current_forwardable
-    logger.debug("publishing message to topic \"platform.sources.event-stream\"...")
-    Sources::Api::Events.raise_event("#{self.class}.destroy", self.as_json, headers)
-    logger.debug("publishing message to topic \"platform.sources.event-stream\"...Complete")
+  IGNORE_RAISE_EVENT_ATTRIBUTES_LIST = %i[availability_status availability_status_error].freeze
+
+  IGNORE_RAISE_EVENT_LIST = {
+    "Application"    => IGNORE_RAISE_EVENT_ATTRIBUTES_LIST,
+    "Authentication" => IGNORE_RAISE_EVENT_ATTRIBUTES_LIST,
+    "Endpoint"       => IGNORE_RAISE_EVENT_ATTRIBUTES_LIST
+  }.freeze
+
+  def ignore_raise_event_for?(attributes)
+    ignore_attribute_list = IGNORE_RAISE_EVENT_LIST[self.class.name]
+    return false unless ignore_attribute_list
+
+    (IGNORE_RAISE_EVENT_ATTRIBUTES_LIST & attributes.map(&:to_sym)).present?
```

Comment:
I think the event should be ignored if there is _only_ one or more attributes from the ignore list. 
@lindgrenj6 what do you think?

Reply:
I'm on the other side, if the application (say cost) PATCH's back into us and updates `extra` at the same time as `availability_status_error` they still are not going to want the update since they are the ones that updated it. They're aware of any changes that go into the application when they're updating the availability status. 

The `availability_status` fields will most likely only ever be updated by the applications themselves so it's safe to ignore them even if they aren't the only fields IMO.

Reply:
ok, it makes sense